sycl/include/sycl/info/info_desc.hpp

```diff
@@ -51,44 +55,84 @@ namespace context {
 } // namespace context
 
 // A.3 Device information descriptors
-enum class device_type : pi_uint64 {
-  cpu = PI_DEVICE_TYPE_CPU,
-  gpu = PI_DEVICE_TYPE_GPU,
-  accelerator = PI_DEVICE_TYPE_ACC,
-  // TODO: figure out if we need all the below in PI
-  custom = PI_DEVICE_TYPE_CUSTOM,
+enum class device_type : uint32_t {
+  cpu = UR_DEVICE_TYPE_CPU,
+  gpu = UR_DEVICE_TYPE_GPU,
+  accelerator = UR_DEVICE_TYPE_FPGA,
+  // TODO: evaluate the need for equivalent UR enums for these types
+  custom,
   automatic,
   host,
-  all = PI_DEVICE_TYPE_ALL
+  all = UR_DEVICE_TYPE_ALL
 };
 
-enum class partition_property : pi_device_partition_property {
+enum class partition_property : intptr_t {
   no_partition = 0,
-  partition_equally = PI_DEVICE_PARTITION_EQUALLY,
-  partition_by_counts = PI_DEVICE_PARTITION_BY_COUNTS,
-  partition_by_affinity_domain = PI_DEVICE_PARTITION_BY_AFFINITY_DOMAIN,
-  ext_intel_partition_by_cslice = PI_EXT_INTEL_DEVICE_PARTITION_BY_CSLICE
+  partition_equally = UR_DEVICE_PARTITION_EQUALLY,
+  partition_by_counts = UR_DEVICE_PARTITION_BY_COUNTS,
+  partition_by_affinity_domain = UR_DEVICE_PARTITION_BY_AFFINITY_DOMAIN,
+  ext_intel_partition_by_cslice = UR_DEVICE_PARTITION_BY_CSLICE
 };
 
-enum class partition_affinity_domain : pi_device_affinity_domain {
+// The old implementation would simply static cast the UR enum to the strongly
+// typed sycl one, but that only worked because the PR "enum" was actually a
+// typedef with some global constexpr values defined in the header. UR defines
+// an actual enum so we need this conversion helper
```

Comment:
I don't think that we need this comment in the code, it references things which won't exist in the repo and therefore provides no value, I think. It is a fine comment to be posted on GH to help reviewers understand the change, though.

Reply:
Good point, deleted that and this review comment should act as a documentation for the future to review this code comment.